lib/httpClient.ts

```diff
@@ -22,6 +23,14 @@ export default abstract class HTTPClient {
             this.client.defaults.headers.common['X-Axiom-Org-Id'] = orgID;
         }
 
+        // We should only retry in the case the status code is >= 500, anything below isn't worth retrying.
+        axiosRetry(this.client, {
+            retryDelay: axiosRetry.exponentialDelay,
+            retryCondition: (error: any) => {
+                return isNetworkError(error) || isRetryableError(error) || (error.response && error.response.status >= 500);
```

Comment:
```suggestion
                return isNetworkError(error) || isRetryableError(error);
```

`isRetryableError` already does the 500 check: https://github.com/softonic/axios-retry/blob/f857c9d72b2e461657f8f117be091c1f0522a74c/es/index.mjs#L28